src/components/rangeslider/draw.js

```diff
@@ -0,0 +1,424 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var d3 = require('d3');
+
+var Plotly = require('../../plotly');
+var Plots = require('../../plots/plots');
+var Axes = require('../../plots/cartesian/axes');
+var Lib = require('../../lib');
+
+var dragElement = require('../dragelement');
+var setCursor = require('../../lib/setcursor');
+
+var constants = require('./constants');
+var rangePlot = require('./range_plot');
+
+
+module.exports = function(gd) {
+    var fullLayout = gd._fullLayout,
+        rangeSliderData = makeRangeSliderData(fullLayout);
+
+    /*
+     * <g container />
+     *  <rect bg />
+     *  < .... range plot />
+     *  <rect mask-min />
+     *  <rect mask-max />
+     *  <rect slidebox />
+     *  <g grabber-min />
+     *      <rect handle-min />
+     *      <rect grabare-min />
+     *  <g grabber-max />
+     *      <rect handle-max />
+     *      <rect grabare-max />
+     *
+     *  ...
+     */
+
+    function keyFunction(axisOpts) {
+        return axisOpts._name;
+    }
+
+    var rangeSliders = fullLayout._infolayer
+        .selectAll('g.' + constants.containerClassName)
+        .data(rangeSliderData, keyFunction);
```

Comment:
As before, there can only be one range slider on a graph at once.

But, the drawing code is now general enough that adding multiple range sliders as a feature would be pretty easy.